### DISCUSSION TITLE
Viewing packets in console

### DISCUSSION THREAD

**Discussion by davthomaspilot** [tagyoureit/nodejs-poolController#461] [Q&A]
dashPanel is showing mixing in the chemistry panel when it should be monitoring status.  Also, there is a ph level high alarm (but ph is ok).

I recall seeing this when I was analyzing the packets last spring.  Maybe I even fixed in a local copy, but now I don't recall.

Is there an entirely different message protocol for the Chemistry messages if REM is used?

I'm planning to switch from Intellichem to REM.  So, I'm not going to open an issue on this, unless you guys would like to spend the time to fix it (maybe I'm the only one who was using it).

**Comment by tagyoureit** [comment]
https://github.com/rstrouse/nodejs-poolController-dashPanel/wiki/Message-Manager is used to view messages.

If the status isn't right, it's definitely worth fixing.  If you can figure it out AND provide a packet capture, that would be ideal.  

> Is there an entirely different message protocol for the Chemistry messages if REM is used?

When we control the chemistry, we control 100% of the logic.  But otherwise, we are still snooping on the OCP/IntelliChem messages.

**Reply by davthomaspilot** [reply]
I mentioned this last June in a Gitter discussion.  Seems "monitoring" and "mixing" states are reversed.  I see "Mixing" in dashPanel all the time.  

Pretty sure I fixed this in a local copy, but I can't recall where the dosingStatus.name are defined.  If you could point me to that, I can compare with what I did in my copy, make the change in my updated copy, and let you know if that fixes it.

**Reply by rstrouse** [reply]
The status is set in `controller/comms/messages/status/IntelliChemStateMessage.ts : line(245)` and the codes are defined in `controller/boards/SystemBoard.ts : line(700)`.

However, you cannot just change the codes this has to match up with the byte(34) on the action 18 message from IntelliChem in IntelliChemStateMessage.ts.  What we need to know is what is the value of byte(34) at the time it is mixing, monitoring, and dosing.  The reason is because these values may be different with chem controller variants.

**Reply by davthomaspilot** [reply]
I have the the values of the bits during mixing, monitoring, and dosing (from last June).  I just want to check what the code is doing.

Thanks!

**Reply by davthomaspilot** [reply]
The bit values are consistent with what I had in my local copy last June.   But, dashPanel currently shows "Mixing" all the time for ph, so some seems fishy.

Maybe I have mixing time set to something ridiculously long... Haven't even checked that.

If that's not the problem, I'll next change mixing time on the Intellichem to something short so I can see dosing, mixing, and monitoring states without have to capture for a long period of time.  ( I have the packets from when I did that last year and the code seems right for that).
